src/queue.h

```diff
@@ -254,6 +254,93 @@ struct {								\
 	_Q_INVALIDATE((elm)->field.le_next);				\
 } while (0)
 
+/*
+ * Singly-linked Tail queue declarations.
+ */
+#define STAILQ_HEAD(name, type)                                                                    \
```

Comment:
I wonder if it is worth, or if using `TAILQ` is good enough...
What do you think ?

Reply:
DPDK requires STAILQ for its function. It loads `sys/queue.h` and since Suricata redefines the file, DPDK misses STAILQ. I know from my earlier experiments that putting DPDK headers at the top of each file kind of worked but if I am not mistaken, it just loaded `sys/queue.h` sooner than Suricata could. I made an attempt to find a better solution. 

For that reason, I added STAILQ implementation to `src/queue.h`. I am currently not aware of possibility to change STAILQ to TAILQ in DPDK. I am, however, open to put it in `#ifdef HAVE_DPDK` clause if required.

Reply:
Fair enough for me :-)

Reply:
@lukashino the manual changes are not required. There is a better way to include the missing elements. Please revisit the same

Reply:
I am not sure what you mean by that. I might be missing something. Could you please elaborate how that can be done in a better way?